### DISCUSSION TITLE
Need to connect an explicit relation inside of a .create() call and I don't have the ID

### DISCUSSION THREAD

**Discussion by giacomocerquone** [prisma/prisma#8755] [Q&A] [answered]
Hi.

I have such a model:

```prisma
model User {
  id              String          @id @default(uuid())
  email           String          @unique
  name            String
  password        String
  group           UsersOnGroups[]
}

model Group {
  id   String          @id @default(uuid())
  name String
  user UsersOnGroups[]
}

model UsersOnGroups {
  admin     Boolean
  user      User    @relation(fields: [userId], references: [id])
  userId String
  group     Group   @relation(fields: [groupId], references: [id])
  groupId   String

  @@id([userId, groupId])
}
```

Where I'm using an explicit relation because I need to add the "role" field.
When I try to create a new user and attach it to a previously created group, I'm writing such a query and **I don't have the userId since I'm creating it:**

```js
prisma.user.create({
      data: {
        name,
        email,
        password: cryptedPwd,
        group: {
          connectOrCreate: {
            where: {
              userId_groupId: {
                groupId: groupId || "", // groupId comes from the request
                userId: WHAT DO I PUT HERE?
              },
            },
            create: {
              admin: true,
              group: {
                create: {
                  name: groupName || "",
                },
              },
            },
          },
        },
      },
    });
```

It seems a duplicate of this: https://github.com/prisma/prisma/issues/3266
So it might be a Prisma bug, but what are the alternatives except using the interactive transaction API?

**Comment by ryands17** [comment, marked as answer]
@giacomocerquone 👋 

Unfortunately there isn't any alternative for this apart from using interactive transactions.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
